src/domains/transaction/hooks/use-transaction-builder.ts

```diff
@@ -16,10 +18,27 @@ const prepareMultiSignature = async (
 		.multiSignature(wallet.multiSignature().all().min, wallet.multiSignature().all().publicKeys),
 });
 
-const prepareLedger = async (input: Services.TransactionInputs, wallet: ProfileContracts.IReadWriteWallet) => ({
-	...input,
-	signatory: await wallet.signatory().ledger(wallet.data().get<string>(ProfileContracts.WalletData.DerivationPath)!),
-});
+const prepareLedger = async (
+	profile: ProfileContracts.IProfile,
+	input: Services.TransactionInputs,
+	wallet: ProfileContracts.IReadWriteWallet,
+	connectFn: ConnectFn,
+	transport: typeof Transport,
+) => {
+	await connectFn(profile, wallet.coinId(), wallet.networkId());
+
+	await wallet.ledger().connect(transport);
+
+	const signatory = await wallet
+		.signatory()
+		.ledger(wallet.data().get<string>(ProfileContracts.WalletData.DerivationPath)!);
+
+	return {
+		...input,
+		signatory,
+		nonce: wallet.nonce().plus(1).toFixed(),
```

Comment:
This can be skipped. The SDK will get the nonce for the address that belongs to the public key of the derivation path and increase it.